os-eap-sso/added/keycloak.sh

```diff
@@ -464,5 +474,97 @@ function get_application_routes {
   elif [ -n "$secureroute" ]; then
     APPLICATION_ROUTES="${secureroute}"
   fi
+}
+
+# Tries to discover the route using the pod's hostname
+# ${1} - secure (true/false)
+discover_route() {
+    local secure=${1}
+    local route=$(python -c "a='${HOSTNAME}'.split('-'); print '-'.join(a[0:len(a)-2])")
```

Comment:
One other note here, python is probably available, but it seems like a heavy requirement for this script. Is it doable with awk?

Reply:
I was also worried for this requirement, but I've tested this script with the EAP base images that have python available. Most RHEL/CentOS base images have python available. For this requirement, I can use awk for sure, but there are another function that uses the json python library to parse k8s API response. I guess that I have to convert that also?

Reply:
Well, I just mentioned in that in case it was an easy switch. As we move forward, things like UBI minimal etc don't have a default exposed python available by default, but we could deal with these issues at that point.

So if its reasonable to do then I'd recommend doing it now, if its a large amount of rework, probably not.

Reply:
@luck3y , aren't the version 3 EAP probes written in Python?

Reply:
@errantepiphany Yeah, but thinking for the future, if we had EAP answer the probes itself we could drop the python dependency. So I'm just asking if its possible .. we certainly have python available now in all of the images we ship.